Fixes issues with pip arguments being used when they shouldn't and not being used when they should

donovan6000 opened the pull request:
#### What does this PR do and why is it necessary?
Makes it so that installing, uninstalling, and updating plugins uses the appropriately provided pip arguments.

#### How was it tested? How can it be tested by the reviewer?
Testing was done by setting the pluginmanager's pip_args setting to various values, like `--user` and nothing, and seeing if I could then install, uninstall, and update a plugin without encountering any unexpected issues.

#### Any background context you want to provide?
Pip wont uninstall a package if it's provided with inapplicable arguments like `--user`, `--ignore-installed`, etc, and it'll just return an error whenever that happens. So using the same user provided pip arguments to install and uninstall a plugin can result in breaking the ability to uninstall plugins.

I felt like the softwareupdater plugin should use the pluginmanager's pip_args setting when it uses pip to update a plugin. Since those pip arguments are what was needed to install the plugin in the first place then they are also needed to install the newest version of that plugin.

#### What are the relevant tickets if any?
None

#### Screenshots (if appropriate)
Not appropriate

#### Further notes
I left in code to make these changes easily scalable to allow adding more arguments that can be blacklisted depending on if pip is installing or uninstalling a package. As of right now it just includes blacklisting the `--user` argument when uninstalling a package since that's what I was most concerned with. I know that there's already a pip_force_user setting that deals with the `--user` argument, but this PR deals with a different issue since it's designed to fix problems with the pip arguments that are provided by the user.

foosel commented on src/octoprint/plugins/softwareupdate/updaters/pip.py:
```diff
@@ -71,6 +72,18 @@ def _log(lines, prefix=None, stream=None):
 	if "dependency_links" in check and check["dependency_links"]:
 		pip_args += ["--process-dependency-links"]
 
+	additional_args = octoprint.plugin.plugin_manager().plugin_implementations["pluginmanager"]._settings.get(["pip_args"])
```
Hm... I don't like that. I know why you are doing it, and the reason is valid, but it couples both plugins and that should be avoided if possible.

I'm wondering if it would be a better idea to maybe extract the whole pip configuration from the plugin manager and into a general platform wide configuration, to be used by both the plugin manager and the software update plugin. Basically taking the extraction of the pip helper one step further. Would necessitate a migration step for the config.

donovan6000 commented:
Updated the PR to use the recommendations in your code review.

I removed the changes to the softwareupdate plugin since you prefer that the plugins remain independent of one another.

foosel commented:
Squashed as new single commit 1b9bfc6 and merged to `devel`. Thanks!
